jms/1.7/modules/ROOT/pages/jms-studio-configuration.adoc

```diff
@@ -0,0 +1,478 @@
+
+
+==== Generic External Libraries
+
+For a *Generic Connection*, provide all the libraries that your connection factory of your implementation requires. Ensure that you all the required dependencies to the application.
+
+In the following example, you configure a generic library for a connection that uses a JMS 2.0 specification, which requires a generic connection with a different client library, like Apache Artemis:
+
+In your `pom.xml` file in Studio, adding the dependency looks as follows:
+
+[source,xml,linenums]
+----
+<dependency>
+    <groupId>org.apache.activemq</groupId>
+    <artifactId>artemis-jms-client-all</artifactId>
+    <version>2.17.0</version>
+</dependency>
+----
+
+To configure these libraries in the Global Element view follow these steps:
+
+. In Studio, navigate to the *Global Elements* tab.
+. Click *Create*.
+. In the filter box, type `jms`, and select *JMS Config*.
+. Click *OK*.
+. In the *Connection* field, select *Generic Connection*.
+. In the *Required Libraries* section that shows the *JMS Client*, click the *Configure...* button to install the dependency.
+. Select any of the install options:
+* *Add recommended library* +
+Installs the recommended library.
+* *Use local file* +
+Browse to a local file for the required engine library and install it.
+* *Add Maven dependency* +
+Browse to the dependency and install it.
+
+=== Configure the JMS Specification
+
+The JMS *Specification* you configure must be supported by the broker implementation associated with the connection, otherwise the connection fails.
+Note that features that are available only in `JMS_2_0` won't work with any other specification.
+
+To configure the JMS specification in Studio, follow these steps:
+
+. In Studio, open the *JMS Config* global element window.
+. In the *Specification* field select one of the following supported specifications:
+* *JMS_1_1 (Default)*
+* *JMS_1_0_2b*
+* *JMS_2_0*
+. Click *OK*.
+
+.JMS Connector specification configuration
+image::jms-spec-config.png[In the Specification field select one of the supported specifications]
+
+In the XML editor, the `specification` configuration looks like this:
+
+[source,xml,linenums]
+----
+<jms:config name="JMS_Config" >
+		<jms:active-mq-connection specification="JMS_1_0_2b" />
+	</jms:config>
+----
+
+=== Configure Connections Caching
+
+To connect with a broker and execute the required operations, the JMS Connector creates multiple sessions of consumers and producers that can be cached and reused in order to increase the performance of the application. JMS Connector enables you to configure the *Caching strategy* field to use when creating new connections. By default, both both consumers and producers are cached and as many instances are preserved as possible in memory concurrently.
+
+It is best not disable the sessions cache by using the *No caching* configuration, which is meant to be used only when a custom connection factory is used that already manages its own cache.
```

Comment:
Because we use "only," let's be more direct:
```suggestion
Do not disable the sessions cache by using the *No caching* configuration, which is used only with a custom connection factory that already manages its own cache.
```